### PR TITLE
added NotFound component + routes

### PR DESCRIPTION
**Pull request by dimoshen**
added NotFoundRedirect page + NotFound component with some custom errror message

### CODE REVIEW

**Comment by Ihor-Karpyn** [comment]
Could u provide screenshots with UI changes?

File: src/App.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ import AccessoriesPage from './pages/AccessoriesPage';
 import MainPage from './pages/MainPage';
 import PhonesPage from './pages/PhonesPage';
 import TabletsPage from './pages/TabletsPage';
+import NotFound from './components/NotFound/NotFound';
```
Comment (ruslanpashkov) [inline src/App.tsx:9]: It's a page, move it in `src/pages/` folder

File: src/App.tsx [hunk 2]
```diff
@@ -28,6 +29,10 @@ const router = createBrowserRouter([
         path: 'accessories',
         element: <AccessoriesPage />,
       },
+      {
+        path: '*',
+        element: <NotFound />,
```
Comment (ruslanpashkov) [inline src/App.tsx:34]:
```suggestion
        element: <NotFoundPage />,
```

File: src/components/NotFound/NotFound.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        </h2>
+        <h4 data-text="Opps! Page not found">
+            Oops! Page not found
+        </h4>
+        <p>
+          Sorry, the page you&apos;re looking for doesn&apos;t exist.
+        </p>
+        <div className="button">
+          <Link to ="/">Return Home</Link>
+        </div>
```
Comment (ruslanpashkov) [inline src/components/NotFound/NotFound.tsx:20]:
```suggestion
        <h2 className="header" data-text="404">
          404
        </h2>

        <h4 data-text="Opps! Page not found">
            Oops! Page not found
        </h4>

        <p>
          Sorry, the page you&apos;re looking for doesn&apos;t exist.
        </p>

        <div className="button">
          <Link to ="/">Return Home</Link>
        </div>
```

File: src/components/NotFound/NotFound.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import { Link } from 'react-router-dom';
+
+const NotFound: FC = () => {
+  return (
+    <div id="error-page">
+      <div className="container">
+        <h2 className="header" data-text="404">
+          404
+        </h2>
+        <h4 data-text="Opps! Page not found">
```
Comment (ruslanpashkov) [inline src/components/NotFound/NotFound.tsx:12]: Do we need this data attribute here?

File: src/components/NotFound/NotFound.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { FC } from 'react';
+import './NotFound.scss';
+import { Link } from 'react-router-dom';
+
+const NotFound: FC = () => {
+  return (
+    <div id="error-page">
```
Comment (ruslanpashkov) [inline src/components/NotFound/NotFound.tsx:7]: Let's use `className` instead. Also we use BEM on our project, so name all classes more appropriately

File: src/pages/NotFoundRedirect.tsx
Comment (ruslanpashkov) [inline src/pages/NotFoundRedirect.tsx:1]: We don't use this file anywhere, let's remove it

File: src/components/NotFound/NotFound.scss
Comment (ruslanpashkov) [inline src/components/NotFound/NotFound.scss:1]: BEM 😭

File: src/components/NotFound/NotFound.scss [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+      50% {
+        background-position: 100% 100%;
+      }
+      75% {
+        background-position: 0% 100%;
+      }
+      100% {
+        background-position: 0% 0%;
+      }
+    }
```
Comment (ruslanpashkov) [inline src/components/NotFound/NotFound.scss:73]: Create a separate `_keyframes.scss` file in `src/styles/utils` and give more descriptive name for keyframe

File: src/components/NotFound/NotFound.scss [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+      a {
+        display: inline-block;
+        margin: 0 10px;
+        text-decoration: none;
+        background-color: #313237;
+        color: #fff;
+        font-size: 14px;
+        padding: 10px 25px;
+        transition: all 0.3s ease;
```
Comment (ruslanpashkov) [inline src/components/NotFound/NotFound.scss:103]: Don't use transitions for all properties, use transitions only for the specific styles

**Review by ruslanpashkov — CHANGES_REQUESTED** [review]
Please refactor the `NotFound.scss` file following best practices from our project

File: src/App.tsx (hunk 1 above)
Comment (dimoshen) [reply src/App.tsx]: removed

File: src/components/NotFound/NotFound.tsx (hunk 4 above)
Comment (dimoshen) [reply src/components/NotFound/NotFound.tsx]: removed. fixed content in the 'after'

File: src/components/NotFound/NotFound.tsx (hunk 5 above)
Comment (dimoshen) [reply src/components/NotFound/NotFound.tsx]: added classNames

File: src/components/NotFound/NotFound.scss
Comment (dimoshen) [reply src/components/NotFound/NotFound.scss:1]: fixed :)

File: src/components/NotFound/NotFound.scss (hunk 6 above)
Comment (dimoshen) [reply src/components/NotFound/NotFound.scss]: fixed

**Pull request merged**